enacts/wat_bal/agronomy.py

```diff
@@ -214,3 +226,122 @@ def soil_plant_water_balance(
             - (planted_since - np.timedelta64(1, "D"))
         )
     return sm, drainage, et_crop, planting_date
+
+
+def api_runoff(
+    daily_rain,
+    api,
+    no_runoff=12.5,
+    api_thresh=DEFAULT_API_THRESHOLD,
+    api_poly=DEFAULT_API_POLYNOMIALS,
+):
+    """Computes Runoff based on an Antecedent Precipitation Index.
+    `runoff` is a polynomial `api_poly` of `daily_rain`.
+    Polynomial is chosen based on API categories defined by `api_thresh`.
+    Additionaly, `runoff` is 0 if it rains less or equal than `no_runoff` ,
+    and negative `runoff` is 0.
+
+    Parameters
+    ----------
+    daily_rain : DataArray
+        daily precipitation
+    api : DataArray
+        an Antecedent Precipitiona Index. Must be same size as daily_rain.
+    no_runoff : DataArray, optional
+        `runoff` is 0 if `daily_rain` is lesser or equal to `no_runoff`
+        (default `no_runoff` =12.5)
+    api_thresh : tuple, optional
```

Comment:
No change needed, but just to be sure you understand what's going on, the argument provided by the user doesn't have to be a tuple, we just needed to make the default value a tuple. This method will work if passed a tuple, list, np.array, xr.Dataset, or anything else that can be iterated over with `for v in ...`. The term that generalizes over that class of objects is "iterable."

Reply:
Then you would describe api_poly as iterable(iterable) ? or iterable of 2 dimensions?

Reply:
Yes, iterable(iterable) or iterable of iterables.